pxr/imaging/hdSt/material.cpp

```diff
@@ -180,15 +255,20 @@ HdStMaterial::Sync(HdSceneDelegate *sceneDelegate,
         std::static_pointer_cast<HdStResourceRegistry>(
             sceneDelegate->GetRenderIndex().GetResourceRegistry());
 
-    HdDirtyBits bits = *dirtyBits;
+    const HdDirtyBits maskedBits = (*dirtyBits & AllDirty);
 
-    if (!(bits & DirtyResource) && !(bits & DirtyParams)) {
+    if (maskedBits == 0) {
         *dirtyBits = Clean;
         return;
     }
 
-    bool needsRprimMaterialStateUpdate = false;
-    bool markBatchesDirty = false;
+#ifdef PXR_MATERIALX_SUPPORT_ENABLED
+    if (maskedBits != DirtyInit) {
+        // This means that the material has been modified since creation,
+        // so the state of the filter task is stale
+        _filterTask.reset();
```

Comment:
This is where we handle cases when the material has changed between creation and the first sync. We'll create a new filter task later, but we can still use the shader produced by parallel codegen, if the changes in parameters haven't changed the shader hash.